### PR TITLE
[CFEP-18] Remove static libs

### LINKED ISSUE #13: benchmarkTargets.cmake points to the static lib
I believe the issue is that in the build script we loop over SHARED being ON then OFF, so the static lib is built second and its CMake files overwrite the shared lib produced ones: https://github.com/conda-forge/benchmark-feedstock/blob/master/recipe/build.sh#L9

Given CFEP-18: https://github.com/conda-forge/cfep/blob/master/cfep-18.md should we just remove the static build altogether?

### PR DESCRIPTION
**Pull request by kkraus14**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes #13 

Removes static libs from build per CFEP-18

### CODE REVIEW

**Comment by kkraus14** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,8 +24,6 @@ test:
   commands:
     - test -f $PREFIX/include/benchmark/benchmark.h  # [unix]
     - if not exist %LIBRARY_INC%\\benchmark\\benchmark.h exit 1  # [win]
-    - test -f $PREFIX/lib/libbenchmark.a  # [unix]
```
Comment (xhochy) [inline recipe/meta.yaml:27]: Make this `test ! -f` so that we check that the static lib is not accidentally included again.

Comment (kkraus14) [reply recipe/meta.yaml:27]: done

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
Thanks @kkraus14

**Pull request merged**